master/internal/api_checkpoint.go

```diff
@@ -73,17 +75,46 @@ func (m *Master) canDoActionOnCheckpoint(
 	return nil
 }
 
+func (m *Master) canDoActionOnCheckpointThroughModel(ctx context.Context, curUser model.User,
+	id uuid.UUID) error {
+	modelIDs, err := m.db.GetModelIDsAssociatedWithCheckpoint(id)
```

Comment:
nitpicking here but for better func arguments / body separation I'd rather indent it as:
```suggestion
func (m *Master) canDoActionOnCheckpointThroughModel(
    ctx context.Context, curUser model.User, id uuid.UUID
) error {
	modelIDs, err := m.db.GetModelIDsAssociatedWithCheckpoint(id)
```